contracts/src/TroveManager.sol

```diff
@@ -464,6 +464,18 @@ contract TroveManager is LiquityBase, ITroveManager, ITroveEvents {
         return _status == Status.active || _status == Status.zombie;
     }
 
+    // Troves with debt adjusted in the same block cannot be liquidated
+    // This is to prevent attacks involving sandwiching a price update to self-liquidate
+    function _isRecent(uint256 _troveId) internal view returns (bool) {
+        if (uint256(Troves[_troveId].lastDebtUpdateTime) >= block.timestamp) return true;
+        address batchAddress = _getBatchManager(_troveId);
+        if (batchAddress != address(0) && uint256(batches[batchAddress].lastDebtUpdateTime) >= block.timestamp) {
```

Comment:
This is interesting. It means that _any_ borrower in a batch can give _all_ Troves in that batch a 1-block liquidation delay, right?
  
But doesn't that mean they could save other Troves from liquidation even though they're below MCR?

e.g. Troves A, B, C in batch, with :

CR_A: 109%
CR_B: 109%
CR_C: 120%

C could slightly adjust his debt every block and save B and A from liquidation, right?

With non-batched Troves it doesnt matter since any adjustment must bring the Trove above MCR, so it really is a liquidation grace period. But with batched Troves, it seems like C can grief liquidation of A and B indefinitely, even though they're below MCR.

Reply:
Good point!
Not sure how to proceed, but I’m leaning towards dismissing the PR.
We need to make sure we block all possible self-triggered liquidations, otherwise it’s not worth making changes.
- If we don’t check last batch debt update, there would be a hole there.
- If we leave it as is, in normal conditions it wouldn’t be an issue, as having to touch trove C (in your example), and beat all liquidation bots to make sure the touch goes first seems unpractical, expensive and unrealistic. But obviously it opens a door for griefing "theoretical" attacks, where you are able to maintain a huge trove (A or B) below MCR, until it could even provoke a shutdown.